tests/http-outgoing.test.js

```diff
@@ -123,3 +125,68 @@ tap.test(
         t.end();
     },
 );
+
+tap.test(
+    'HttpOutgoing() - call .writeEarlyHints() - should not all repsonse.writeEarlyHints() if no assets have been set',
+    (t) => {
+        t.plan(1);
+        let count = 0;
+        const incoming = new HttpIncoming(
+            {},
+            {
+                writeEarlyHints() {
+                    count++;
+                },
+            },
+        );
+        const outgoing = new HttpOutgoing(
+            RESOURCE_OPTIONS,
+            REQ_OPTIONS,
+            incoming,
+        );
+        outgoing.writeEarlyHints();
+
+        t.equal(count, 0);
+    },
+);
+
+tap.test(
+    'HttpOutgoing() - call .writeEarlyHints() - should call repsonse.writeEarlyHints()',
+    (t) => {
+        t.plan(3);
+        let result;
+        const incoming = new HttpIncoming(
+            {},
+            {
+                writeEarlyHints(options, cb) {
+                    result = options;
+                    cb();
+                },
+            },
+        );
+        const outgoing = new HttpOutgoing(
+            RESOURCE_OPTIONS,
+            REQ_OPTIONS,
+            incoming,
+        );
+        outgoing.js = [
+            new AssetJs({ value: 'https://assets.foo.com/scripts.js' }),
+        ];
+        outgoing.css = [
+            new AssetCss({ value: 'https://assets.foo.com/styles.js' }),
+        ];
+
+        outgoing.writeEarlyHints(() => {
+            t.equal(result.link.length, 2);
+            t.equal(
+                result.link[0],
+                '<https://assets.foo.com/scripts.js>; type=application/javascript; rel=preload; as=script; asset-type=style',
+            );
+            t.equal(
+                result.link[1],
+                '<https://assets.foo.com/styles.js>; type=text/css; rel=preload; as=style; asset-type=style',
```

Comment:
```suggestion
                '<https://assets.foo.com/styles.js>; type=text/css; rel=preload; as=style',
```

Reply:
Right you are, I've fixed by removing the asset tagging from the utils package and moving it to the podlet so that using toHeader here no longer generates the asset-type= bits